server/data/data-library/src/main/java/org/apache/james/rrt/lib/MappingImpl.java

```diff
@@ -99,9 +103,18 @@ public String getErrorMessage() {
     }
 
     @Override
-    public String getAddress() {
-        Preconditions.checkState(getType() == Type.Address);
-        return mapping;
+    public Optional<MailAddress> getAddress(ValidationMode validationMode) {
+        if (validationMode == ValidationMode.Strict) {
+            Preconditions.checkState(getType() == Type.Address || getType() == Type.Forward);
+        }
+        if (type != Type.Address && type!= Type.Forward) {
```

Comment:
you should factorize the condition and rely on the "mode" to decide what the action is